### ISSUE TITLE
GEOS Chem 12.9.3 ERROR: Error encountered in wet deposition!  at SAFETY  in area RAINOUT

### ISSUE THREAD

**Issue by XunZhangNJU** [geoschem/geos-chem#1721]
### Name and Institution (Required)

Name: XunZhang
Institution: Nanjing University

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question
When run GEOS Chem 12.9.3, GCclassic version. Compiled "merra2_4x5_standard", Time range: 20160701 - 20160801. The models always report these errors：
```console
K, IC, Q(K,IC):    5  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):   12  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):   10  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    8  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):   10  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    9  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    9  76           NaN
Infinity in DO_CLOUD_CONVECTION!
Infinity in DO_CLOUD_CONVECTION!
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    4  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    4  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    5  76           NaN
K, IC, Q(K,IC):    6  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    4  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    6  76           NaN
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    9  76           NaN
Infinity in DO_CLOUD_CONVECTION!
Infinity in DO_CLOUD_CONVECTION!
K, IC, Q(K,IC):    5  76           NaN
Infinity in DO_CLOUD_CONVECTION!
.......
 LOST        :   0.000000000000000E+000
 ALPHA2      :   0.000000000000000E+000
 RAINFRAC    :   1.560697383153847E-003
 Spc(I,J,:N) :                      NaN                     NaN
                     NaN                     NaN                     NaN
                     NaN                     NaN                     NaN
                     NaN                     NaN                     NaN
                     NaN  0.000000000000000E+000  0.000000000000000E+000
forrtl: warning (406): fort: (1): In call to SAFETY, an array temporary was created for argument

  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
forrtl: warning (406): fort: (1): In call to SAFETY, an array temporary was created for argument 

  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
..............
 0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
WETDEP: ERROR at    1  30   9 for species   76 in area RAINOUT
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
forrtl: warning (406): fort: (1): In call to SAFETY, an array temporary was created for argument
............
===============================================================================
 MASS_WASH   :   0.000000000000000E+000
===============================================================================
GEOS-Chem ERROR: Error encountered in wet deposition!
 PDOWN       :   2.179046987293986E-007
 -> at SAFETY (in module GeosCore/wetscav_mod.F90)
 

Please provide as much detail as possible. Always include the GEOS-Chem version number and any relevant configuration and log files.
Version: 12.9.3
merra2_4x5_standard
CODE_DIR     : ./CodeDir
  CODE_BRANCH  : GC_12.9.3
  LAST_COMMIT  : Fix typo in ncdf_mod.F90 from previous commit
  COMMIT_DATE  : Wed Aug 5 17:38:19 2020 -0400
  VERSION      : 12.9.3
  VERSION_TAG  : GC_12.9.3
  TRACEBACK    : y
  BOUNDS       : n
  DEBUG        : y
  FPE          : n
  OMP          : y
  CHEM         : Standard
  TOMAS12      : n
  TOMAS15      : n
  TOMAS30      : n
  TOMAS40      : n
  APM          : n
  RRTMG        : n
  TAU_PROF     : n
  BPCH_DIAG    : n
  COMPILER     : ifort 19.0.1.144
  Datetime     : 2023/03/24 09:19
#####################
```
![Quicker_20230324_161852](https://user-images.githubusercontent.com/22512680/227463947-7df5c8c9-c1a5-4f16-978d-4b3fb2e2bab1.png)

[GC.log](https://github.com/geoschem/geos-chem/files/11059681/GC.log)
[HEMCO_Config.txt](https://github.com/geoschem/geos-chem/files/11059727/HEMCO_Config.txt)
[HEMCO_Diagn.txt](https://github.com/geoschem/geos-chem/files/11059728/HEMCO_Diagn.txt)
[HISTORY.txt](https://github.com/geoschem/geos-chem/files/11059729/HISTORY.txt)

**Comment by yantosca** [comment]
Thanks for writing. @XunZhangNJU. We have some general debugging guidance at geos-chem.readthedocs.io, https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html

You are getting some NaN (not-a-number) values in your code.  This may indicate a division by zero somewhere.  

Also note, GEOS-Chem version 12 is no longer officially supported.  We would recommend updating to a more recent version to take advantage of several bug fixes that have been implemented since 12.9.3 was released.

**Comment by yantosca** [comment]
This error could also be caused by corrupted met field or emissions data.  You might want to check the meterology files to see if you can visualize them with ncview or a similar netCDF program.
